assets/prometheus-k8s/rules.yaml

```diff
@@ -380,6 +380,13 @@ spec:
       record: node_role_os_version_machine:cpu_capacity_sockets:sum
     - expr: clamp_max(sum(alertmanager_notifications_total),1)
       record: cluster:alertmanager_routing_enabled:max
+    - alert: ClusterOOMWarning
+      annotations:
+        message: Cluster is experiencing out of memory (OOM) kills
+      expr: sum(max(kube_pod_container_status_last_terminated_reason{reason="OOMKilled",job="kube-state-metrics"}))
```

Comment:
Do we care about all namespaces, including user namespaces? I would limit this to openshift namespaces like we do for other pod alerts. https://github.com/openshift/cluster-monitoring-operator/blob/2d04b3a7bb69c8333f132d39f91488d60b02a7b4/assets/prometheus-k8s/rules.yaml#L588

Reply:
critical namespaces makes sense. A customer can configure additional alerts for their namespaces. +1